tests/lib/bin/apt-tool

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python3
+
+import sys
+
+import apt
+
+
+def checkpoint():
+    pkgs = set()
+    cache = apt.Cache()
+    for pkg in cache:
+        if pkg.is_installed:
+            pkgs.add(pkg.name)
+    for name in sorted(pkgs):
+        print(name)
+
+
+def restore(fname):
+    desired = set([line.strip() for line in open(fname)])
+            
+    cache = apt.Cache()
+    for pkg in cache:
+        if pkg.is_installed and not pkg.name in desired:
+            print("removing", pkg)
+            pkg.mark_delete(auto_fix=False)
+        if not pkg.is_installed and pkg.name in desired:
+            print("installing", pkg)
+            pkg.mark_install(auto_fix=False, auto_install=False)
+    cache.commit()
+
+
+if __name__ == "__main__":
+    if sys.argv[1] == "checkpoint":
+        checkpoint()
+    elif sys.argv[1] == "restore":
+        restore(sys.argv[2])
```

Comment:
else `exit(1)` to ensure we don't miss typos in the tests?

Reply:
I'll follow up with argparse-based command line handling so that this part is accounted for. If you agree with that we can land and Ill iterate.

Reply:
Sure.